### PR TITLE
Admin user management

### PR DESCRIPTION
**Pull request by notmd**
part of #1022 
- Allow updating the show_on_leaderboard field.
- Add raw JSON of the user object.
- Add a new user message table.
- Also fixed style issue: hard to see reaction count when the user also reacted.
- Rename MessageTable to MessageConversation.

UI
image
image

### CODE REVIEW

File: website/src/components/Messages/AdminMessageTable.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    cell: ({ getValue }) => {
+      return <DateDiff>{getValue()}</DateDiff>;
+    },
+  }),
+  columnHelper.accessor((row) => row.id, {
+    header: "Actions",
+    cell: ({ getValue }) => (
+      <DataTableAction
+        as={NextLink}
+        href={ROUTES.ADMIN_MESSAGE_DETAIL(getValue())}
```
Comment (notmd) [inline website/src/components/Messages/AdminMessageTable.tsx:90]: this page doesn't exists yet, I will add it in the next PR

File: website/src/components/DataTable/useCursorPagination.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  };
+
+  const toNextPage = (data: undefined | { prev?: string; next?: string }) => {
+    setPagination({
+      cursor: data?.next || "",
+      direction: "forward",
+    });
+  };
+
+  const resetCursor = () => setPagination((old) => ({ ...old, cursor: "" }));
```
Comment (AbdBarho) [inline website/src/components/DataTable/useCursorPagination.ts:32]: `useCallback` for these 3 functions.

File: website/src/components/Messages/AdminMessageTable.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
+import { Avatar } from "@chakra-ui/avatar";
+import { Badge, Flex } from "@chakra-ui/layout";
+import { Tooltip } from "@chakra-ui/react";
+import { createColumnHelper } from "@tanstack/table-core";
+import { formatDistanceToNow, formatISO9075 } from "date-fns";
```
Comment (AbdBarho) [inline website/src/components/Messages/AdminMessageTable.tsx:5]: I wonder if this is something that `next-i18n` already provides.

**Review by AbdBarho — COMMENTED** [review]
Thanks! it would be great if you could make upcoming PRs smaller than this, it is easier to review.

File: website/src/components/Messages/AdminMessageTable.tsx (hunk 3 above)
Comment (fozziethebeat) [reply website/src/components/Messages/AdminMessageTable.tsx:5]: From my knowledge, the answer is no.  I'm pretty sure we'd have to use some new library to do this so date-fns seems good enough

**Review by fozziethebeat — COMMENTED** [review]
Since @AbdBarho left some good comments, I have nothing else to add.  Looks fine after those changes.

**Review by fozziethebeat — APPROVED** [review]
By special request by Andreas I'm going to approve and merge this.  Maybe make the cleanups in a follow up PR?

File: website/src/components/DataTable/useCursorPagination.ts (hunk 2 above)
Comment (notmd) [reply website/src/components/DataTable/useCursorPagination.ts:32]: These 3 function will need a wrapper function to pass data, so useCallback has no effect here.

**Pull request merged**